Issue on compiling 'SEARCHING' is not a member of 'esphome::esp32_ble_tracker::ClientState'

Hi!

I get a strange error on compiling:

``In file included from src/esphome/components/awox_mesh/mesh_connection.cpp:5:
src/esphome/components/awox_mesh/mesh_connection.h: In member function 'virtual void esphome::awox_mesh::MeshConnection::set_state(esphome::esp32_ble_tracker::ClientState)':
src/esphome/components/awox_mesh/mesh_connection.h:121:44: error: 'SEARCHING' is not a member of 'esphome::esp32_ble_tracker::ClientState'
  121 |       case esp32_ble_tracker::ClientState::SEARCHING:
      |                                            ^~~~~~~~~
src/esphome/components/awox_mesh/mesh_connection.cpp: In member function 'void esphome::awox_mesh::MeshConnection::connect_to(esphome::awox_mesh::FoundDevice*)':
src/esphome/components/awox_mesh/mesh_connection.cpp:53:51: error: 'SEARCHING' is not a member of 'esphome::esp32_ble_tracker::ClientState'
   53 |   this->set_state(esp32_ble_tracker::ClientState::SEARCHING);
      |                                                   ^~~~~~~~~
In file included from src/esphome/components/awox_mesh/awox_mesh_mqtt.h:9,
                 from src/esphome/components/awox_mesh/awox_mesh.h:15,
                 from src/esphome/components/awox_mesh/awox_mesh.cpp:7:
src/esphome/components/awox_mesh/mesh_connection.h: In member function 'virtual void esphome::awox_mesh::MeshConnection::set_state(esphome::esp32_ble_tracker::ClientState)':
src/esphome/components/awox_mesh/mesh_connection.h:121:44: error: 'SEARCHING' is not a member of 'esphome::esp32_ble_tracker::ClientState'
  121 |       case esp32_ble_tracker::ClientState::SEARCHING:
      |                                            ^~~~~~~~~
In file included from src/esphome/components/awox_mesh/awox_mesh_mqtt.h:9,
                 from src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:14:
src/esphome/components/awox_mesh/mesh_connection.h: In member function 'virtual void esphome::awox_mesh::MeshConnection::set_state(esphome::esp32_ble_tracker::ClientState)':
src/esphome/components/awox_mesh/mesh_connection.h:121:44: error: 'SEARCHING' is not a member of 'esphome::esp32_ble_tracker::ClientState'
  121 |       case esp32_ble_tracker::ClientState::SEARCHING:
      |                                            ^~~~~~~~~
Compiling .pioenvs/esphome-web-26d444/src/esphome/components/esp32/gpio.cpp.o
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp: In lambda function:
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:90:58: warning: 'ArduinoJson::V742HB22::JsonObject ArduinoJson::V742HB22::JsonObject::createNestedObject(const TString&) [with TString = std::__cxx11::basic_string<char>]' is deprecated: use obj[key].to<JsonObject>() instead [-Wdeprecated-declarations]
   90 |           JsonObject connection = root.createNestedObject("connection_" + std::to_string(i));
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from .piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson.hpp:39,
                 from .piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson.h:9,
                 from src/esphome/components/json/json_util.h:11,
                 from src/esphome/core/string_ref.h:11,
                 from src/esphome/core/entity_base.h:5,
                 from src/esphome/components/mqtt/mqtt_component.h:10,
                 from src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:10:
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:218:14: note: declared here
  218 |   JsonObject createNestedObject(const TString& key) {
      |              ^~~~~~~~~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp: In lambda function:
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:178:53: warning: 'ArduinoJson::V742HB22::JsonObject ArduinoJson::V742HB22::JsonObject::createNestedObject(TChar*) [with TChar = const char]' is deprecated: use obj[key].to<JsonObject>() instead [-Wdeprecated-declarations]
  178 |           JsonObject color = root.createNestedObject("color");
      |                              ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:211:14: note: declared here
  211 |   JsonObject createNestedObject(TChar* key) {
      |              ^~~~~~~~~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp: In lambda function:
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:212:59: warning: 'ArduinoJson::V742HB22::JsonObject ArduinoJson::V742HB22::JsonObject::createNestedObject(TChar*) [with TChar = const char]' is deprecated: use obj[key].to<JsonObject>() instead [-Wdeprecated-declarations]
  212 |           JsonObject device_info = root.createNestedObject(MQTT_DEVICE);
      |                                    ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:211:14: note: declared here
  211 |   JsonObject createNestedObject(TChar* key) {
      |              ^~~~~~~~~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp: In lambda function:
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:234:59: warning: 'ArduinoJson::V742HB22::JsonObject ArduinoJson::V742HB22::JsonObject::createNestedObject(TChar*) [with TChar = const char]' is deprecated: use obj[key].to<JsonObject>() instead [-Wdeprecated-declarations]
  234 |           JsonObject device_info = root.createNestedObject(MQTT_DEVICE);
      |                                    ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:211:14: note: declared here
  211 |   JsonObject createNestedObject(TChar* key) {
      |              ^~~~~~~~~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp: In lambda function:
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:256:59: warning: 'ArduinoJson::V742HB22::JsonObject ArduinoJson::V742HB22::JsonObject::createNestedObject(TChar*) [with TChar = const char]' is deprecated: use obj[key].to<JsonObject>() instead [-Wdeprecated-declarations]
  256 |           JsonObject device_info = root.createNestedObject(MQTT_DEVICE);
      |                                    ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:211:14: note: declared here
  211 |   JsonObject createNestedObject(TChar* key) {
      |              ^~~~~~~~~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp: In lambda function:
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:278:59: warning: 'ArduinoJson::V742HB22::JsonObject ArduinoJson::V742HB22::JsonObject::createNestedObject(TChar*) [with TChar = const char]' is deprecated: use obj[key].to<JsonObject>() instead [-Wdeprecated-declarations]
  278 |           JsonObject device_info = root.createNestedObject(MQTT_DEVICE);
      |                                    ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:211:14: note: declared here
  211 |   JsonObject createNestedObject(TChar* key) {
      |              ^~~~~~~~~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp: In lambda function:
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:303:59: warning: 'ArduinoJson::V742HB22::JsonObject ArduinoJson::V742HB22::JsonObject::createNestedObject(TChar*) [with TChar = const char]' is deprecated: use obj[key].to<JsonObject>() instead [-Wdeprecated-declarations]
  303 |           JsonObject device_info = root.createNestedObject(MQTT_DEVICE);
      |                                    ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:211:14: note: declared here
  211 |   JsonObject createNestedObject(TChar* key) {
      |              ^~~~~~~~~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp: In lambda function:
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:341:56: warning: 'ArduinoJson::V742HB22::JsonArray ArduinoJson::V742HB22::JsonObject::createNestedArray(TChar*) const [with TChar = const char]' is deprecated: use obj[key].to<JsonArray>() instead [-Wdeprecated-declarations]
  341 |         JsonArray availability = root.createNestedArray(MQTT_AVAILABILITY);
      |                                  ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:197:13: note: declared here
  197 |   JsonArray createNestedArray(TChar* key) const {
      |             ^~~~~~~~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:342:68: warning: 'ArduinoJson::V742HB22::JsonObject ArduinoJson::V742HB22::JsonArray::createNestedObject() const' is deprecated: use add<JsonObject>() instead [-Wdeprecated-declarations]
  342 |         auto availability_topic_1 = availability.createNestedObject();
      |                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
In file included from .piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson.hpp:65:
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/compatibility.hpp:140:19: note: declared here
  140 | inline JsonObject JsonArray::createNestedObject() const {
      |                   ^~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:344:68: warning: 'ArduinoJson::V742HB22::JsonObject ArduinoJson::V742HB22::JsonArray::createNestedObject() const' is deprecated: use add<JsonObject>() instead [-Wdeprecated-declarations]
  344 |         auto availability_topic_2 = availability.createNestedObject();
      |                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/compatibility.hpp:140:19: note: declared here
  140 | inline JsonObject JsonArray::createNestedObject() const {
      |                   ^~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:346:68: warning: 'ArduinoJson::V742HB22::JsonObject ArduinoJson::V742HB22::JsonArray::createNestedObject() const' is deprecated: use add<JsonObject>() instead [-Wdeprecated-declarations]
  346 |         auto availability_topic_3 = availability.createNestedObject();
      |                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/compatibility.hpp:140:19: note: declared here
  140 | inline JsonObject JsonArray::createNestedObject() const {
      |                   ^~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:357:55: warning: 'ArduinoJson::V742HB22::JsonArray ArduinoJson::V742HB22::JsonObject::createNestedArray(TChar*) const [with TChar = const char]' is deprecated: use obj[key].to<JsonArray>() instead [-Wdeprecated-declarations]
  357 |         JsonArray color_modes = root.createNestedArray("supported_color_modes");
      |                                 ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:197:13: note: declared here
  197 |   JsonArray createNestedArray(TChar* key) const {
      |             ^~~~~~~~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:363:57: warning: 'ArduinoJson::V742HB22::JsonArray ArduinoJson::V742HB22::JsonObject::createNestedArray(TChar*) const [with TChar = const char]' is deprecated: use obj[key].to<JsonArray>() instead [-Wdeprecated-declarations]
  363 |           JsonArray effect_list = root.createNestedArray(MQTT_EFFECT_LIST);
      |                                   ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:197:13: note: declared here
  197 |   JsonArray createNestedArray(TChar* key) const {
      |             ^~~~~~~~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:387:57: warning: 'ArduinoJson::V742HB22::JsonObject ArduinoJson::V742HB22::JsonObject::createNestedObject(TChar*) [with TChar = const char]' is deprecated: use obj[key].to<JsonObject>() instead [-Wdeprecated-declarations]
  387 |         JsonObject device_info = root.createNestedObject(MQTT_DEVICE);
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:211:14: note: declared here
  211 |   JsonObject createNestedObject(TChar* key) {
      |              ^~~~~~~~~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:389:62: warning: 'ArduinoJson::V742HB22::JsonArray ArduinoJson::V742HB22::JsonObject::createNestedArray(TChar*) const [with TChar = const char]' is deprecated: use obj[key].to<JsonArray>() instead [-Wdeprecated-declarations]
  389 |         JsonArray identifiers = device_info.createNestedArray(MQTT_DEVICE_IDENTIFIERS);
      |                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:197:13: note: declared here
  197 |   JsonArray createNestedArray(TChar* key) const {
      |             ^~~~~~~~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp: In lambda function:
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:461:56: warning: 'ArduinoJson::V742HB22::JsonArray ArduinoJson::V742HB22::JsonObject::createNestedArray(TChar*) const [with TChar = const char]' is deprecated: use obj[key].to<JsonArray>() instead [-Wdeprecated-declarations]
  461 |         JsonArray availability = root.createNestedArray(MQTT_AVAILABILITY);
      |                                  ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:197:13: note: declared here
  197 |   JsonArray createNestedArray(TChar* key) const {
      |             ^~~~~~~~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:462:68: warning: 'ArduinoJson::V742HB22::JsonObject ArduinoJson::V742HB22::JsonArray::createNestedObject() const' is deprecated: use add<JsonObject>() instead [-Wdeprecated-declarations]
  462 |         auto availability_topic_1 = availability.createNestedObject();
      |                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/compatibility.hpp:140:19: note: declared here
  140 | inline JsonObject JsonArray::createNestedObject() const {
      |                   ^~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:464:68: warning: 'ArduinoJson::V742HB22::JsonObject ArduinoJson::V742HB22::JsonArray::createNestedObject() const' is deprecated: use add<JsonObject>() instead [-Wdeprecated-declarations]
  464 |         auto availability_topic_2 = availability.createNestedObject();
      |                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/compatibility.hpp:140:19: note: declared here
  140 | inline JsonObject JsonArray::createNestedObject() const {
      |                   ^~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:466:68: warning: 'ArduinoJson::V742HB22::JsonObject ArduinoJson::V742HB22::JsonArray::createNestedObject() const' is deprecated: use add<JsonObject>() instead [-Wdeprecated-declarations]
  466 |         auto availability_topic_3 = availability.createNestedObject();
      |                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/compatibility.hpp:140:19: note: declared here
  140 | inline JsonObject JsonArray::createNestedObject() const {
      |                   ^~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:477:55: warning: 'ArduinoJson::V742HB22::JsonArray ArduinoJson::V742HB22::JsonObject::createNestedArray(TChar*) const [with TChar = const char]' is deprecated: use obj[key].to<JsonArray>() instead [-Wdeprecated-declarations]
  477 |         JsonArray color_modes = root.createNestedArray("supported_color_modes");
      |                                 ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:197:13: note: declared here
  197 |   JsonArray createNestedArray(TChar* key) const {
      |             ^~~~~~~~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:483:57: warning: 'ArduinoJson::V742HB22::JsonArray ArduinoJson::V742HB22::JsonObject::createNestedArray(TChar*) const [with TChar = const char]' is deprecated: use obj[key].to<JsonArray>() instead [-Wdeprecated-declarations]
  483 |           JsonArray effect_list = root.createNestedArray(MQTT_EFFECT_LIST);
      |                                   ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:197:13: note: declared here
  197 |   JsonArray createNestedArray(TChar* key) const {
      |             ^~~~~~~~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:507:57: warning: 'ArduinoJson::V742HB22::JsonObject ArduinoJson::V742HB22::JsonObject::createNestedObject(TChar*) [with TChar = const char]' is deprecated: use obj[key].to<JsonObject>() instead [-Wdeprecated-declarations]
  507 |         JsonObject device_info = root.createNestedObject(MQTT_DEVICE);
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:211:14: note: declared here
  211 |   JsonObject createNestedObject(TChar* key) {
      |              ^~~~~~~~~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:509:62: warning: 'ArduinoJson::V742HB22::JsonArray ArduinoJson::V742HB22::JsonObject::createNestedArray(TChar*) const [with TChar = const char]' is deprecated: use obj[key].to<JsonArray>() instead [-Wdeprecated-declarations]
  509 |         JsonArray identifiers = device_info.createNestedArray(MQTT_DEVICE_IDENTIFIERS);
      |                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:197:13: note: declared here
  197 |   JsonArray createNestedArray(TChar* key) const {
      |             ^~~~~~~~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp: In member function 'void esphome::awox_mesh::AwoxMeshMqtt::process_incomming_command(esphome::awox_mesh::MeshDestination*, ArduinoJson::V742HB22::JsonObject)':
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:538:23: warning: 'bool ArduinoJson::V742HB22::JsonObject::containsKey(const TString&) const [with TString = char [14]; typename ArduinoJson::V742HB22::detail::enable_if<ArduinoJson::V742HB22::detail::IsString<TString>::value, int>::type <anonymous> = 0]' is deprecated: use obj[key].is<T>() instead [-Wdeprecated-declarations]
  538 |   if (root.containsKey("fade_duration")) {
      |       ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:168:8: note: declared here
  168 |   bool containsKey(const TString& key) const {
      |        ^~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:543:23: warning: 'bool ArduinoJson::V742HB22::JsonObject::containsKey(const TString&) const [with TString = char [15]; typename ArduinoJson::V742HB22::detail::enable_if<ArduinoJson::V742HB22::detail::IsString<TString>::value, int>::type <anonymous> = 0]' is deprecated: use obj[key].is<T>() instead [-Wdeprecated-declarations]
  543 |   if (root.containsKey("color_duration")) {
      |       ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:168:8: note: declared here
  168 |   bool containsKey(const TString& key) const {
      |        ^~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:548:23: warning: 'bool ArduinoJson::V742HB22::JsonObject::containsKey(const TString&) const [with TString = char [6]; typename ArduinoJson::V742HB22::detail::enable_if<ArduinoJson::V742HB22::detail::IsString<TString>::value, int>::type <anonymous> = 0]' is deprecated: use obj[key].is<T>() instead [-Wdeprecated-declarations]
  548 |   if (root.containsKey("color")) {
      |       ~~~~~~~~~~~~~~~~^~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:168:8: note: declared here
  168 |   bool containsKey(const TString& key) const {
      |        ^~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:563:23: warning: 'bool ArduinoJson::V742HB22::JsonObject::containsKey(const TString&) const [with TString = char [11]; typename ArduinoJson::V742HB22::detail::enable_if<ArduinoJson::V742HB22::detail::IsString<TString>::value, int>::type <anonymous> = 0]' is deprecated: use obj[key].is<T>() instead [-Wdeprecated-declarations]
  563 |   if (root.containsKey("brightness") && !root.containsKey("color_temp") &&
      |       ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:168:8: note: declared here
  168 |   bool containsKey(const TString& key) const {
      |        ^~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:563:58: warning: 'bool ArduinoJson::V742HB22::JsonObject::containsKey(const TString&) const [with TString = char [11]; typename ArduinoJson::V742HB22::detail::enable_if<ArduinoJson::V742HB22::detail::IsString<TString>::value, int>::type <anonymous> = 0]' is deprecated: use obj[key].is<T>() instead [-Wdeprecated-declarations]
  563 |   if (root.containsKey("brightness") && !root.containsKey("color_temp") &&
      |                                          ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:168:8: note: declared here
  168 |   bool containsKey(const TString& key) const {
      |        ^~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:564:24: warning: 'bool ArduinoJson::V742HB22::JsonObject::containsKey(const TString&) const [with TString = char [6]; typename ArduinoJson::V742HB22::detail::enable_if<ArduinoJson::V742HB22::detail::IsString<TString>::value, int>::type <anonymous> = 0]' is deprecated: use obj[key].is<T>() instead [-Wdeprecated-declarations]
  564 |       (root.containsKey("color") || mesh_destination->color_mode)) {
      |        ~~~~~~~~~~~~~~~~^~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:168:8: note: declared here
  168 |   bool containsKey(const TString& key) const {
      |        ^~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:574:30: warning: 'bool ArduinoJson::V742HB22::JsonObject::containsKey(const TString&) const [with TString = char [11]; typename ArduinoJson::V742HB22::detail::enable_if<ArduinoJson::V742HB22::detail::IsString<TString>::value, int>::type <anonymous> = 0]' is deprecated: use obj[key].is<T>() instead [-Wdeprecated-declarations]
  574 |   } else if (root.containsKey("brightness")) {
      |              ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:168:8: note: declared here
  168 |   bool containsKey(const TString& key) const {
      |        ^~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:585:23: warning: 'bool ArduinoJson::V742HB22::JsonObject::containsKey(const TString&) const [with TString = char [11]; typename ArduinoJson::V742HB22::detail::enable_if<ArduinoJson::V742HB22::detail::IsString<TString>::value, int>::type <anonymous> = 0]' is deprecated: use obj[key].is<T>() instead [-Wdeprecated-declarations]
  585 |   if (root.containsKey("color_temp")) {
      |       ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:168:8: note: declared here
  168 |   bool containsKey(const TString& key) const {
      |        ^~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:597:23: warning: 'bool ArduinoJson::V742HB22::JsonObject::containsKey(const TString&) const [with TString = char [7]; typename ArduinoJson::V742HB22::detail::enable_if<ArduinoJson::V742HB22::detail::IsString<TString>::value, int>::type <anonymous> = 0]' is deprecated: use obj[key].is<T>() instead [-Wdeprecated-declarations]
  597 |   if (root.containsKey("effect")) {
      |       ~~~~~~~~~~~~~~~~^~~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:168:8: note: declared here
  168 |   bool containsKey(const TString& key) const {
      |        ^~~~~~~~~~~
src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp:617:23: warning: 'bool ArduinoJson::V742HB22::JsonObject::containsKey(const TString&) const [with TString = char [6]; typename ArduinoJson::V742HB22::detail::enable_if<ArduinoJson::V742HB22::detail::IsString<TString>::value, int>::type <anonymous> = 0]' is deprecated: use obj[key].is<T>() instead [-Wdeprecated-declarations]
  617 |   if (root.containsKey("state")) {
      |       ~~~~~~~~~~~~~~~~^~~~~~~~~
.piolibdeps/esphome-web-26d444/ArduinoJson/src/ArduinoJson/Object/JsonObject.hpp:168:8: note: declared here
  168 |   bool containsKey(const TString& key) const {
      |        ^~~~~~~~~~~
*** [.pioenvs/esphome-web-26d444/src/esphome/components/awox_mesh/mesh_connection.cpp.o] Error 1
*** [.pioenvs/esphome-web-26d444/src/esphome/components/awox_mesh/awox_mesh.cpp.o] Error 1
*** [.pioenvs/esphome-web-26d444/src/esphome/components/awox_mesh/awox_mesh_mqtt.cpp.o] Error 1``

--- a/components/awox_mesh/mesh_connection.cpp
+++ b/components/awox_mesh/mesh_connection.cpp
@@ -50,7 +50,7 @@ static int get_product_code(unsigned char part1, unsigned char part2) { return i
 
 void MeshConnection::connect_to(FoundDevice *found_device) {
   this->set_address(found_device->device.address_uint64());
-  this->set_state(esp32_ble_tracker::ClientState::SEARCHING);
+  this->set_state(esp32_ble_tracker::ClientState::DISCOVERED);
   this->found_device = found_device;
   this->found_device->connected = true;
 

--- a/components/awox_mesh/mesh_connection.cpp
+++ b/components/awox_mesh/mesh_connection.cpp
@@ -49,6 +49,8 @@ static std::string encrypt(std::string key, std::string data) {
 static int get_product_code(unsigned char part1, unsigned char part2) { return int(part2); }
 
 void MeshConnection::connect_to(FoundDevice *found_device) {
+  this->set_address(found_device->device.address_uint64());
+  this->set_state(esp32_ble_tracker::ClientState::IDLE);
   this->found_device = found_device;
   this->found_device->connected = true;
 

--- a/components/awox_mesh/mesh_connection.cpp
+++ b/components/awox_mesh/mesh_connection.cpp
@@ -49,8 +49,6 @@ static std::string encrypt(std::string key, std::string data) {
 static int get_product_code(unsigned char part1, unsigned char part2) { return int(part2); }
 
 void MeshConnection::connect_to(FoundDevice *found_device) {
-  this->set_address(found_device->device.address_uint64());
-  this->set_state(esp32_ble_tracker::ClientState::DISCOVERED);
   this->found_device = found_device;
   this->found_device->connected = true;
 

--- a/components/awox_mesh/mesh_connection.h
+++ b/components/awox_mesh/mesh_connection.h
@@ -107,39 +107,34 @@ class MeshConnection : public esp32_ble_client::BLEClientBase {
     std::string state = "";
     switch (st) {
       case esp32_ble_tracker::ClientState::INIT:
-
         state = "INIT";
         break;
-      case esp32_ble_tracker::ClientState::DISCONNECTING:
 
+      case esp32_ble_tracker::ClientState::DISCONNECTING:
         state = "DISCONNECTING";
         break;
-      case esp32_ble_tracker::ClientState::IDLE:
 
+      case esp32_ble_tracker::ClientState::IDLE:
         state = "IDLE";
         break;
-      case esp32_ble_tracker::ClientState::SEARCHING:
 
-        state = "SEARCHING";
-        break;
       case esp32_ble_tracker::ClientState::DISCOVERED:
-
         state = "DISCOVERED";
         break;
-      case esp32_ble_tracker::ClientState::READY_TO_CONNECT:
 
+      case esp32_ble_tracker::ClientState::READY_TO_CONNECT:
         state = "READY_TO_CONNECT";
         break;
-      case esp32_ble_tracker::ClientState::CONNECTING:
 
+      case esp32_ble_tracker::ClientState::CONNECTING:
         state = "CONNECTING";
         break;
-      case esp32_ble_tracker::ClientState::CONNECTED:
 
+      case esp32_ble_tracker::ClientState::CONNECTED:
         state = "CONNECTED";
         break;
-      case esp32_ble_tracker::ClientState::ESTABLISHED:
 
+      case esp32_ble_tracker::ClientState::ESTABLISHED:
         state = "ESTABLISHED";
         break;
 

